client/my-sites/migrate/section-migrate.jsx

```diff
@@ -92,23 +112,51 @@ class SectionMigrate extends Component {
 	};
 
 	startMigration = () => {
-		const { sourceSiteId, targetSiteId } = this.props;
+		const { sourceSiteId, targetSiteId, targetSite } = this.props;
 
 		if ( ! sourceSiteId || ! targetSiteId ) {
 			return;
 		}
 
+		const planSlug = get( targetSite, 'plan.product_slug' );
+		if (
+			planSlug &&
+			! this._startedMigrationFromCart &&
+			! planHasFeature( planSlug, FEATURE_UPLOAD_THEMES_PLUGINS )
+		) {
+			this.goToCart();
+			return;
+		}
+
 		this.setMigrationState( { migrationStatus: 'backing-up' } );
 
 		wpcom
 			.startMigration( sourceSiteId, targetSiteId )
 			.then( () => this.updateFromAPI() )
 			.catch( error => {
-				const { message = '' } = error;
-				this.setMigrationState( { migrationStatus: 'error', errorMessage: message } );
+				const { code = '', message = '' } = error;
+
+				if ( 'no_supported_plan' === code ) {
+					this.goToCart();
+					return;
+				}
+
+				this.setMigrationState( {
+					migrationStatus: 'error',
+					errorMessage: message,
+				} );
 			} );
 	};
 
+	goToCart = () => {
+		const { sourceSite, targetSiteSlug } = this.props;
+		const sourceSiteSlug = get( sourceSite, 'slug' );
+
+		page(
+			`/checkout/${ targetSiteSlug }/business?redirect_to=/migrate/from/${ sourceSiteSlug }/to/${ targetSiteSlug }%3Fstart%3Dtrue`
```

Comment:
Handy. I was worried that it'd be more difficult to do this. Happy to see that this is pretty straight-forward.

Reply:
Very handy!